android/java/org/chromium/chrome/browser/vpn/utils/InAppPurchaseWrapper.java

```diff
@@ -52,10 +51,20 @@ public SkuDetails getSkuDetails(String sku) {
     public static final List<String> SUBS_SKUS = new ArrayList<>(
             Arrays.asList(NIGHTLY_MONTHLY_SUBSCRIPTION, NIGHTLY_YEARLY_SUBSCRIPTION));
 
-    public static InAppPurchaseWrapper getInstance() {
-        if (sInAppPurchaseWrapper == null) sInAppPurchaseWrapper = new InAppPurchaseWrapper();
+    private static volatile InAppPurchaseWrapper sInAppPurchaseWrapper;
+    private static Object mutex = new Object();
+
+    private InAppPurchaseWrapper() {}
 
-        return sInAppPurchaseWrapper;
+    public static InAppPurchaseWrapper getInstance() {
+        InAppPurchaseWrapper result = sInAppPurchaseWrapper;
+        if (result == null) {
+            synchronized (mutex) {
```

Comment:
same here, maybe we can simplify by using `InAppPurchaseWrapper.class` instead of the `Object mutex`